### ISSUE TITLE
How to create generalized weight matrix from multiple linear regression

### ISSUE THREAD

**Issue by moonj94** [statsmodels/statsmodels#7252]
#### Describe the bug

This isn't really a bug, but rather a request for some assistance. ( I didn't know which category to submit an issue). 

So I am dealing with some EEG data in tasks A and B. I wish to do multiple linear regression between them. However, these data are set up as shape (300 trials, 64 channels, 128 samples). I tried concatenating the trials along the time dimension to do MLR, however the resultant accuracy and fit were poor. 

Is there a better way in which to create a 'generalized weight matrix' from MLR that is an accurate fit?

#### Code Sample, a copy-pastable example if possible

```python
        # %% split up training data into separate classes
        # concatenate SP and CS across trials along the time dimension
        # perform statsmodel.api.WLS/OLS/or some other regression type
        # extract regression weights
        # apply to training, validation, and testing data of CS
        spbidx=np.where(label_train==6)
        spoidx=np.where(label_train==7)
        csbidx=np.where(label_train==8)
        csoidx=np.where(label_train==9)        
        rstidx=np.where(label_train==5)
        
        spb=X_train[spbidx[0],:,:]
        spo=X_train[spoidx[0],:,:]
        csb=X_train[csbidx[0],:,:]
        cso=X_train[csoidx[0],:,:]
        sp=np.concatenate((spb,spo),0)
        cs=np.concatenate((csb,cso),0)
        spv=sp.reshape([128*sp.shape[0],64])
        csv=cs.reshape([128*cs.shape[0],64])       
        
        d=spv
        x=csv
        x=sm.add_constant(x)
        est=sm.WLS(d,x).fit()
        csr=np.matmul(x,est.params)
        
        plt.subplot(2,1,1)
        plt.plot(d[0:1000,1])
        plt.subplot(2,1,2)
        plt.plot(csr[0:1000,1])
        plt.show()
```
<details>

**Note**: As you can see, there are many issues on our GitHub tracker, so it is very possible that your issue has been posted before. Please check first before submitting so that we do not have to handle and close duplicates.

**Note**: Please be sure you are using the latest released version of `statsmodels`, or a recent build of `master`. If your problem has been fixed in an unreleased version, you might be able to use `master` until a new release occurs. 

**Note**: If you are using a released version, have you verified that the bug exists in the master branch of this repository? It helps the limited resources if we know problems exist in the current master so that they do not need to check whether the code sample produces a bug in the next release.

</details>

If the issue has not been resolved, please file it in the issue tracker.

#### Output of ``import statsmodels.api as sm; sm.show_versions()``

<details>

[paste the output of ``import statsmodels.api as sm; sm.show_versions()`` here below this line]

</details>

**Comment by josef-pkt** [comment]
OLS/WLS are only for univariate response variable, endog y. So, `d` in your model needs to be 1-dim.

300*64*128 = 2457600

What's the structure of your explanatory variables `x`?
If that is supposed to have many columns, explanatory variables, then you will likely run out of memory using plain OLS, WLS.

Based on your reshaping, `csv` has only one explanatory variable, In that case you could ravel/flatten `d` and `csv`. If you want to include any dummies for fixed effects in trials, channels or samples, then `x` it will get much larger.

**Comment by moonj94** [comment]
the structure of x is = (128 samples, 64 channels) for single trials. If I'm concatenating across trials then dimension 0 is 128 samples * 300 trials. 

I have tried doing 2-dimensional regression using WLS and OLS where x=(128*64) and d=(128*64) and it seems to regress just fine (I think). Are you telling me that the current statsmodel doesn't do multiple linear regression? If so, what is the appropriate way to do it?

csv is of shape (128 * 300, 64)

Yes so what I want to do is construct a weight matrix that generalizes across trials and incorporates information from other trials.

**Comment by josef-pkt** [comment]
standard terminology: 
multiple regression is one y, several x
multivariate regression is several y, several x, x might be the same or different for each y.

I guess with "weight matrix" you mean the design matrix `x`, not weights for each observations to reflect varying variances.

Statsmodels doesn't have general multivariate OLS/WLS,  (there's an incomplete class that supports MANOVA)

MixedLM and GEE support one way clustered/panel data
The linearmodels package by Kevin supports panel data i.e, endog is 2d, but if x differ, then it also needs to work with a flattened model like OLS, with some kind of sparse structure or kronecker products for x.

> csv is of shape (128 * 300, 64)

Does this mean only one explanatory variable for each one of the 128x300x64 observation points.
Is `csv` the task indicator, A, B, from your description?

> and incorporates information from other trials.

What information do you need to include from other trials?
in other words, how are the different 128x300x64 observation points linked? Do they have common parameters, common intercept, ... ?

e.g. 
simplest model; `y_t,c,s = constant + trial_indicator_t,c,s`       common effect of trial A,B across al trials, channels and samples, t,c,s 
or `y_t,c,s = constant_t + trial_indicator_t,c,s`  common effect of trial A versus B, but channel specific intercept

**Comment by moonj94** [comment]
Sorry I meant multivariate regression. Is there a way to run sm.WLS/OLS with a third dimension such as trials? I want the model to incorporate information from all trials not just one set of multivariate data.
